components/style/stylearc.rs

```diff
@@ -71,6 +71,40 @@ pub struct Arc<T: ?Sized> {
     ptr: *mut ArcInner<T>,
 }
 
+/// An Arc that is known to be uniquely owned
+///
+/// This lets us build arcs that we can mutate before
+/// freezing, without needing to change the allocation
+pub struct UniqueArc<T: ?Sized>(Arc<T>);
```

Comment:
Let's move this out of here, to keep this as close as the original `arc.rs` as possible.